lib/screens/player_screen.dart

```diff
@@ -173,3 +163,107 @@ class _PlayerScreenContent extends StatelessWidget {
     );
   }
 }
+
+enum PlayerHideable {
+  bigPlayButton(14, 14),
+  queueButton(0, 27),
+  progressSlider(0, 14),
+  twoLineTitle(0, 27),
+  codecInfo(0, 20),
+  loopShuffleButtons(96, 0);
+
+  final double width;
+  final double height;
+
+  const PlayerHideable(this.width, this.height);
+}
+
+class PlayerHideableController {
+  final verticalHideOrder = [
+    PlayerHideable.bigPlayButton,
+    PlayerHideable.queueButton,
+    PlayerHideable.codecInfo,
+    PlayerHideable.twoLineTitle,
+    PlayerHideable.progressSlider
+  ];
+
+  List<PlayerHideable> _visible = [];
+  Size? _target;
+
+  void setSizePortrait(Size size) {
+    _visible = List.from(PlayerHideable.values);
+
+    var maxPadding =
+        FinampSettingsHelper.finampSettings.playerScreenCoverMinimumPadding + 8;
+    var targetHeight =
+        size.height + size.width * ((maxPadding / 100.0) * 2 - 1);
+    var targetWidth = size.width;
+    _target = Size(targetWidth, targetHeight);
+
+    if (_getSize().width >= targetWidth) {
+      _visible.remove(PlayerHideable.bigPlayButton);
+      if (_getSize().width >= targetWidth) {
+        _visible.add(PlayerHideable.bigPlayButton);
+        _visible.remove(PlayerHideable.loopShuffleButtons);
+        if (_getSize().width >= targetWidth) {
+          _visible.remove(PlayerHideable.bigPlayButton);
+        }
+      }
+    }
+
+    for (var element in verticalHideOrder) {
+      if (_getSize().height < targetHeight) {
+        return;
+      }
+      _visible.remove(element);
+    }
+  }
+
+  void setSizeLandscape(Size size) {
+    _visible = List.from(PlayerHideable.values);
+
+    var targetHeight = size.height - 5;
+    if (_getSize().width >= size.width * 0.6) {
+      _visible.remove(PlayerHideable.bigPlayButton);
+      if (_getSize().width >= size.width * 0.6) {
+        _visible.add(PlayerHideable.bigPlayButton);
+        _visible.remove(PlayerHideable.loopShuffleButtons);
+        if (_getSize().width >= size.width * 0.6) {
+          _visible.remove(PlayerHideable.bigPlayButton);
+        }
+      }
+    }
+    // Prevent allocating extra space between 50% and 60% of width if we're just
+    // going to shrink the play button anyway.
+    if (_getSize().height >= targetHeight) {
+      _visible.remove(PlayerHideable.bigPlayButton);
+    }
+    var targetWidth = max(_getSize().width, size.width / 2);
+    _target = Size(targetWidth, targetHeight);
+
+    for (var element in verticalHideOrder) {
+      if (_getSize().height < targetHeight) {
+        return;
+      }
+      _visible.remove(element);
+    }
+  }
+
+  Size _getSize() {
+    double height = 162;
+    double width = 144;
+    for (var element in _visible) {
+      height += element.height;
+      width += element.width;
+    }
+    return Size(width, height);
+  }
+
+  Size getTarget() {
+    return _target ?? _getSize();
+  }
+
+  bool shown(PlayerHideable element) {
```

Comment:
I'd prefer `isShown()`, `shouldBeShown()` or `canBeShown()` to prevent confusing it with `show()`

Reply:
changed to shouldShow()